rust/lance-encoding/src/encodings/physical/block_compress.rs

```diff
@@ -103,8 +156,18 @@ impl BufferCompressor for ZstdBufferCompressor {
     }
 
     fn decompress(&self, input_buf: &[u8], output_buf: &mut Vec<u8>) -> Result<()> {
-        let source = Cursor::new(input_buf);
-        zstd::stream::copy_decode(source, output_buf)?;
+        if input_buf.is_empty() {
+            output_buf.clear();
```

Comment:
I don't think we should call `clear` here.  We use this path in 2.1 to decompress multiple values (and each one should add to the same `output_buf` so then a later empty value might clear an earlier decompressed value)